openmdao/core/notebook_utils.py

```diff
@@ -0,0 +1,130 @@
+"""Checking for interactive notebook mode."""
+import sys
+import importlib
+import inspect
+
+from IPython.display import display, HTML, Code
+
+from openmdao.utils.general_utils import simple_warning
+
+try:
+    from tabulate import tabulate
+except ImportError:
+    tabulate = None
+
+colab = 'google.colab' in sys.modules
+
+
+def _get_object_from_reference(reference):
+    """
+    Return object of given reference path.
+
+    Parameters
+    ----------
+    reference : str
+        Dot path of desired class.
+
+    Returns
+    -------
+    str
+        Object of the given class.
+    """
+    split = reference.split('.')
+    right = []
+    module = None
+    while split:
+        try:
+            module = importlib.import_module('.'.join(split))
+            break
+        except ModuleNotFoundError:
+            right.append(split.pop())
+    if module:
+        for entry in reversed(right):
+            module = getattr(module, entry)
+    return module
+
+
+def display_source(reference, hide_doc_string=False):
+    """
+    Return the source code of the given reference path.
+
+    Parameters
+    ----------
+    reference : str
+        Dot path of desired class or function.
+    hide_doc_string : bool
+        Option to hide the docstring.
+
+    Returns
+    -------
+    IPython.display
+        Source code of the given class or function.
+    """
+    obj = inspect.getsource(_get_object_from_reference(reference))
+
+    if hide_doc_string:
+        obj = obj.split('"""')
```

Comment:
Not sure if 'reference' can ever be a class here, but if so, getsource would return all of its source including methods and their docstrings, and your split on '"""' and rejoin on '.' would print all of the function docstrings with '.'s replacing the '"""'s.  If reference can't be a class, maybe put something in the docstring stating that (or stating that reference must be a function or whatever).

Reply:
Fixed